### ISSUE TITLE
IMPORTANT NOTE: GoRevo - Evo 2 Revo

### ISSUE THREAD

**Issue by modxuser** [evolution-cms/evolution#466]
BobRay from the MODX forum created a tool some time back that helps to convert Evo sites to be used in Revo.

**With that said:**

It should be made clear that with the removal of so-called "old snippets" that anyone wishing to convert an Evo site in the future will not be able to use GoRevo if they convert all their calls, as it was built to convert Ditto, Wayfinder etc.

It will still possibly work to some degree, but it won't be anywhere near what it can do now.

This is not meant as a dig at the new code, this is simply to make clear to those who are still considering updating to Revo to do it before they convert all their calls to the new snippets, because once sites start using the new snippets, converting will become much more complicated, unless BobRay updates GoRevo, which I personally can't imagine.

This is probably only interesting for front-end users, developers will probaby convert their own sites if and when required.

**Comment by modxuser** [comment]
On a personal note:

I will be looking back into converting my sites, if I have to learn new snippets and have to get my custom code re-coded I may as well do so in Revo.

Why; because at the moment I see the Revo community is growing in comparison to the ever shrinking Evo community

Plus the problems that are continuing with language, documentation, general support and IMHO a lack of interaction between the development side and the community.

**Comment by Pathologic** [comment]
But what's the need of transfering sites to Revo if you are satisfied with Ditto, Wayfinder and other trash?

**Comment by modxuser** [comment]
Your use of "Trash" shows exactly what I no-longer agree with

**Comment by pmfx** [comment]
@modxuser Well, I simply do not agree :) 

EVO development is better than ever and community is active and growing. Sure docs needs improvement and it is happening right now, but most of what you said simply **does not reflect reality**. 

I don't want to compare EVO and Revo situation, but Revo is starting to have a little difficult times with **slow development, still not clear plans for "MODX 3" and it's strong bound to a really old ExtJS 3** framework. 

And if you look closer you will notice that **EVO community has a lot more influence on its development**. Just look at a pile of Revo issues and ideas (over 500) and pull request (over 40) that are waiting on GitHub. This may change soon (hopefully), but this is how it looks now.

Look, I don't want to say that Revo is a bad choice. **Revo is a great CMS** and I believe it has a bright future ahead. That's no doubt. I wish it all the best (I'm a Revo user too!), **but it is not only candy and sweets**, you know.

Also, if you think about it, converting EVO sites to Revo can take **a lot more time than you think** (depending on site complexity) when compared to learning new snippets (DocLister, I suppose) that were developed to boost our sites performance and give us more freedom and flexibility. Is it worth it?

**Comment by modxuser** [comment]
@pmfx 

Is it worth it = yes

Not being a developer leaves me to pay for what I need - I have paid out multiple thousands over the years for custom code based on core snippets that have been in Evo for years, which are now no-longer being supported / about to have support stopped

You are right about Revo - neither of the two (Evo or Revo) are in my opinion failsafe solutions, but I feel Revo is now for me the better option

I appreciate that you have yourself contributed tons of time, as well as others have, to improve on Evo and help keep it up to scratch with the latest technology. I too have helped where I could, but as mentioned, I now see more benefits in Revo than Evo.

I never thought I would say it, but I think 1.4.0 will be my last Evo update

**Comment by dmi3yy** [comment]
if you think that people who are engaged in the development of a roar more than those who develops evo then I can surprise you:
http://take.ms/vXlhx
And you can see the same statistic on video from november 
https://modcasts.video/en/videos/evolution-cms-live-wo-modx

Yes EVO 2 REVO no more work, because REVO havent alternative like DocLister and FormLister with prepare functionals.

**Comment by dmi3yy** [comment]
@modxuser if you want you can use Revo, i not see any problem. 

Now we not MODX we are Evolution CMS
and this step gave us the opportunity to develop without looking back at the REVO. 

Based on my experience sites on evo: 
- work faster both in the admin panel and frontend 
- are created faster and easier and also cheaper for the end customer. 

Yes, there are some problems with the documentation , official site, but Rome was not built in one day.

**Comment by dmi3yy** [comment]
After MODXpo2017 I was convinced that I made the right choice aside EVO. 

ExtJs 3 realy ? is great code? For what year?? https://docs.sencha.com/extjs/6.2.0/ 

Now actual version 6.2 not 3 . 

And in REVO havent plans for this year change this. And I doubt that such plans will be even next year

**Comment by dmi3yy** [comment]
for those who want to understand why EVO:
https://modcasts.video/en/videos/evolution-cms-live-wo-modx

**Comment by dmi3yy** [comment]
If you not like Evolution CMS not use it 
Last MODX EVO is 1.2.2 use it and after evo2revo ;) 

We cant evolution without changes, its imposible.

**Comment by modxuser** [comment]
@dmi3yy 

This is not a discussion about whether or not Revo is better than Evo, or vice versa. This is simply to inform those people who are still interested in updating their Evo sites to Revo that the GoRevo code won't work with new snippets like DocLister, DLMenu, DLCrumbs, Formlister etc.

**Comment by modxuser** [comment]
@dmi3yy 

You are absolutely right, Rome wasn't built in a day, Rome was built over many years with lots of people, but with one common language

Asking people to move from long standing code to new stuff without supporting documentation in a common language isn't an option in my opinion.

I would have thought that the Russian snippet documentation would have been translated before releasing it to the international community. DocLister has been translated, but it is complicated to understand and you are expected to be able to, in some circumstances, code in PHP when using the snippet.

The argument / discussion that keeps arising about translations isn't in my opinion relevant, simply because Evo has an international community and has always used International English, and as such should be carried on in this form

Code from international users to date has always been translated by the code creators themselves and / or in cooperation with others into International English

The problem here is simple, there are no (to my knowledge) non-russian speakers who can translate Russian in the current Evo community, which leaves the onus on the Russian speaking community to translate themselves

IMHO if Evo is to stay an international project further updates on the core should be put on hold (other than security fixes) and priority has to be the creation of the new English documentation site. If this is carried out successfuly I am sure more PHP developers and end-users will be attracted to Evo.

**Comment by dmi3yy** [comment]
@modxuser If thay wont use DocLister, DLMenu, DLCrumbs, Formlister they decide what REVO it's not for him.

**Comment by modxuser** [comment]
@dmi3yy that's obvious, but that's not what I am saying here

**Comment by dmi3yy** [comment]
@modxuser  you can help with translate i not see problems: 
i do new doc site: http://docs.evo.im/en/
Anyone can help with docs on GithHub https://github.com/evolution-cms/docs

I think that I should solve more global problems
such as updating the repository and creating the website evo.im. 

Yes, I can translate all the documentation into English, but unfortunately I am not a native speaker and so the translation will be very bad. In the same way, anyone can translate with  google translate.

In this case, I'm ready to prompt you to explain that wherever. Ie instead of this large dialogue, you could already translate 3-5 pages of documentation :) So it would be more useful.

**Comment by modxuser** [comment]
@dmi3yy 

You know for a fact that I have no problems doing documentation - the current English site was built by me after weeks of researching broken sites with bits and pieces of different docs.

So please dont patronise me

**Comment by dmi3yy** [comment]
Then I especially do not understand the essence of this topic

then that evo went its own way and the fact that automatic updates to REVO will not be understandable even after the release of 1.3.0

Then what's the question? in that we have less community? or is it that we prevent the MODX from developing REVO?

I do not understand what I need to do :) 

As far as the priority of tasks is concerned, I know what is needed to give a good impetus to development, but for this it is necessary to prepare everything and this is a lot of work that I do a little bit and the community actively helps me in this.

**Comment by modxuser** [comment]
> Then I especially do not understand the essence of this topic

As already mentioned, the topic was to state that those who are still interested in updating to Revo wont be able to use GoRevo if they convert their calls to the new code

I probably shouldn't have added my personal thoughts, as that seems to have misled the topic

**Comment by fourroses666** [comment]
Would be easier to go from MODX Evolution to Evolution CMS then going to REVO.

**Comment by dmi3yy** [comment]
@modxuser  ok now understand :). but i think use evo2revo not good 
on revo have better code like pdoTools

**Comment by modxuser** [comment]
@dmi3yy I think you misunderstand

The GoRevo tool converts Ditto, Wayfinder etc. calls to equivalents in Revo, like PDOTools

The topic header "Evo 2 Revo" is simply a short form for "convert from Evo to Revo"

**Comment by modxuser** [comment]
@fourroses666 Marc, you are probably right, but at the moment there is no tool to convert one to the other, other than GoRevo

**Comment by Pathologic** [comment]
Closed because Revo related issues are discussed here: https://github.com/modxcms/revolution/issues

**Comment by modxuser** [comment]
Thanks for the link - I didn't know where the Revo GitHub is

**Comment by risingisland** [comment]
> Then I especially do not understand the essence of this topic

From my understanding, the problem is that for the past 5+ years, from Modx v.95 all the way up to Evo v1.36, Ditto, Wayfinder, etc. have been an essential part of the program with support and documentation in English.

A new and improved replacement for these is more than welcome, but there was no transition period nor information in English to help with this transition.

We were only offered a take it or leave it decision. 

I dont think anyone is arguing that the new DL replacements aren't better, only that there was no clear information on how to migrate or a transition period to do so.

At least that has been my take on the original portion of this thread.

**Comment by modxuser** [comment]
@dmi3yy 

Please review this post: https://forums.modx.com/thread/103432/important-latest-evo-update-1-4-0

If anything is not correct please let me know ASAP

**Comment by dmi3yy** [comment]
We are not stop support old extras like Ditto Wayfinder ets 
they are stil support and will support. They work on MODx api and we will not change this 

for example in Revo all solution in Extras and its normal 
in core all extras have eng version. Maybe some one havent eng docs

**Comment by modxuser** [comment]
Then obviously there is a lack of communication here - I was under the impression that the old snippets were no-longer being worked on and will not be worked on in the future

Are you now "officially saying" that all of the old snippets will fully work in all future releases of Evolution CMS ?

**Comment by dmi3yy** [comment]
Yes they will work.
I see no reason why you need to stop supporting.
In fact, the question is whether they will work on new versions of PHP, for example, eFrom at 7.2 does not work anymore, but at the same time no problems on the Evo 1.4.0 and php 7.1 

Also, no one will introduce a new example in the ditto functionality such as "prepare". 

Also, Ditto and other have official repo https://github.com/extras-evolution/Ditto
https://github.com/extras-evolution/
like other snippet where you can write issues and get answer and support )

**Comment by modxuser** [comment]
Ah, OK.

So if a Snippet no-longer works in a certain version of PHP it wont be officialy updated to work in that version or future PHP releases ?

**Comment by modxuser** [comment]
Example:

If Ditto, Wayfinder etc. stop working in PHP 7.2 / 7.3 etc. they won't be officialy updated, only updated if someone (possibly) from the community updates it via its repo

**Comment by dmi3yy** [comment]
Like in REVO
now different updates for  CORE and EXTRAS ) 

after i finish work on new extras all snippets migrate to extras ) but want do better then this in REVO 

https://github.com/extras-evolution 
There are more than 100 additions in the repository, in which everything works the same as in the main repository of the system. Ie, there is an opportunity to send a pulrequest or leave a note that something is not working. For example, I recently helped to run on 1.4.0 easy2gallery but i not used in my project ) 

I do not understand why you decided that we threw out the old one and do not support it anymore? Yes, we strongly recommend that you update because it positively affects the speed of the site + gives more opportunities, but no one said that we do not support the old solutions. I also have a lot of projects where old solutions are used and they also need to be maintained. There are simply priorities for tasks.

It is also always possible to stimulate $ in order to spend time for solving important tasks for you.

**Comment by dmi3yy** [comment]
For start write issues in Ditto and Wayfinder where and what not working : ) 
now: 
 Ditto  - 0 open issue
 Wayfinder - 0 open issue

**Comment by dmi3yy** [comment]
https://github.com/extras-evolution/Wayfinder/issues
https://github.com/extras-evolution/Ditto/issues

**Comment by modxuser** [comment]
This is the problem when it comes to a breakdown in language

The impression was given that the old snippets were no-longer being supported and when comments such as "Trash" are used, they emphasise this

I would never have started this thread if the impression had not been there

**Comment by dmi3yy** [comment]
I understand that the language barrier interferes with communication, but unfortunately it happened so, here the task is to be a little bit more loyal to each other and just once again to ask everything correctly and clearly understood

**Comment by Nicola1971** [comment]
@modxuser 

I had hoped the extrascheck widget had solved this discussion.
Older versions of the snippets are not compatible, **not new ones**. You Just need to update :)

![check14](https://user-images.githubusercontent.com/7342798/35442425-8680ac04-02a7-11e8-9df1-d98d202b791f.png)

There is no reason to to fear that support for these snippets will be abandoned.

Simply, the new features will be developed on the more recent code of Doclister and other the new extras, but everything that has always worked in Ditto will continue to work.

I've have updated most of my sites, and only on tattocms i've updated ditto to doclister and just for the test.

**Comment by modxuser** [comment]
I have updated the forum post to replicate what has been discussed here
